### DISCUSSION TITLE
How do I import a json dataset added manually?

### DISCUSSION THREAD

**Discussion by guille-amarilla** [GetDKAN/dkan#4096] [Q&A]
Hi, how can I import a json dataset added manually? I follow the documentation for "How to add a Dataset" through GUI. In the documentation says to run drush cron to start the import. It worked with csv dataset added, but somehow for json does not work. Can you help me with it? Thanks

**Comment by janette** [comment]
@guille-amarilla to clarify, your dataset resource (distribution) is a json file? or you are trying to create a dataset using the API and json formatted metadata?

If the data itself is json, then DKAN would just provide a download link to it. Only csv data resources can be imported to the datastore and made available via the API.

**Reply by guille-amarilla** [reply]
Yes, it's a json file.

I understand now. Thanks for the respond.

In the case I want to create a dataset using API and json formatted metadata? It would show like a table formatted like the imported csv file? And if it would, how can I accomplish that? Thanks so much for the help so far.

**Comment by janette** [comment]
@guille-amarilla in that case checkout this page in the docs about creating a dataset via the metastore API: https://dkan.readthedocs.io/en/latest/user-guide/guide_dataset.html#post--api-1-metastore-schemas-dataset-items

This will create the dataset and generate a queue to localize the remote file, then another queue is generated to run the import so you can run those queues manually with drush, or just run cron twice. Then you should see the data preview table on your dataset page. If you do not see it, check /admin/dkan/datastore/status to see if the import is reporting an error.
